code/modules/projectiles/projectile/magic.dm

```diff
@@ -24,27 +24,46 @@
 	damage_type = BRUTE
 	nodamage = 0
 
+	//explosion values
+	var/exp_heavy = 0
+	var/exp_light = 2
+	var/exp_flash = 3
+	var/exp_fire = 2
+
 /obj/item/projectile/magic/fireball/Range()
 	var/turf/T1 = get_step(src,turn(dir, -45))
 	var/turf/T2 = get_step(src,turn(dir, 45))
+	var/turf/T3 = get_step(src,dir)
 	var/mob/living/L = locate(/mob/living) in T1 //if there's a mob alive in our front right diagonal, we hit it.
 	if(L && L.stat != DEAD)
-		Bump(L) //Magic Bullet #teachthecontroversy
+		Bump(L,1) //Magic Bullet #teachthecontroversy
 		return
 	L = locate(/mob/living) in T2
 	if(L && L.stat != DEAD)
-		Bump(L)
+		Bump(L,1)
+		return
+	L = locate(/mob/living) in T3
+	if(L && L.stat != DEAD)
+		Bump(L,1)
 		return
 	..()
 
 /obj/item/projectile/magic/fireball/on_hit(target)
 	. = ..()
 	var/turf/T = get_turf(target)
-	explosion(T, -1, 0, 2, 3, 0, flame_range = 2)
+	explosion(T, -1, exp_heavy, exp_light, exp_flash, 0, flame_range = exp_fire)
 	if(ismob(target)) //multiple flavors of pain
 		var/mob/living/M = target
 		M.take_overall_damage(0,10) //between this 10 burn, the 10 brute, the explosion brute, and the onfire burn, your at about 65 damage if you stop drop and roll immediately
 
+/obj/item/projectile/magic/fireball/infernal
+	name = "infernal fireball"
+	exp_heavy = -1
```

Comment:
I do not believe it does. Nothing odd happened in testing. I am not exactly sure what the difference between -1 and 0 is for explosion code. I am using the same values that were provided in the original spell.